### PR TITLE
dnn onnx: add mod

### PR DESCRIPTION
**Pull request by fengyuentau**
Resolves https://github.com/opencv/opencv/issues/23174

TODO:

- [x] enable some conformance tests
- [x] add backends
    - [x] CANN
    - [x] OpenVINO
    - [x] CUDA

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/onnx/onnx_importer.cpp [hunk 1]
```diff
@@ -2830,6 +2830,11 @@ void ONNXImporter::parseElementWise(LayerParams& layerParams, const opencv_onnx:
 
     layerParams.type = "NaryEltwise";
     layerParams.set("operation", toLowerCase(node_proto.op_type()));
+    if (node_proto.op_type() == "Mod") {
+        if (layerParams.has("fmod") && layerParams.get<int>("fmod") == 1) {
```
Comment (dkurt) [inline modules/dnn/src/onnx/onnx_importer.cpp:2834]:
```suggestion
        if (layerParams.get<int>("fmod", 0)) {
```

File: modules/dnn/src/layers/nary_eltwise_layers.cpp [hunk 2]
```diff
@@ -923,6 +943,12 @@ class NaryEltwiseLayerImpl CV_FINAL : public NaryEltwiseLayer
             node = std::make_shared<ngraph::op::v1::GreaterEqual>(inp0, inp1);
         else if (op == OPERATION::LESS_EQUAL)
             node = std::make_shared<ngraph::op::v1::LessEqual>(inp0, inp1);
+        // Ideally we should do this but int32 internal blobs are converted to float32 data type in inference.
+        // FIXME: enable this when internal blobs support int32 data type
+        // else if (op == OPERATION::IMOD)
+        //     node = std::make_shared<ngraph::op::v1::FloorMod>(inp0, inp1);
```
Comment (dkurt) [inline modules/dnn/src/layers/nary_eltwise_layers.cpp:949]: Maybe `ngraph::op::Convert` can help here?

File: modules/dnn/src/onnx/onnx_importer.cpp (hunk 1 above)
Comment (fengyuentau) [reply modules/dnn/src/onnx/onnx_importer.cpp]: Done.

File: modules/dnn/src/layers/nary_eltwise_layers.cpp (hunk 2 above)
Comment (fengyuentau) [reply modules/dnn/src/layers/nary_eltwise_layers.cpp:949]: It should work. Let me have a try on this.

Comment (fengyuentau) [reply modules/dnn/src/layers/nary_eltwise_layers.cpp]: Done.

**Comment by asmorkalov** [comment]
cc @LaurentBerger

**Review by dkurt — APPROVED** [review]

**Pull request merged**